### PR TITLE
Using local variable to avoid null ref

### LINKED ISSUE #2708: NullRef when using short timeouts for http binding
This NullRef doesn't happen on the timer thread (so if the user code handles the exception it won't crash the process) but it is so frequent that it blocks entire stress testing of other fixes for time out issues.

### PR DESCRIPTION
**Pull request by KKhurin**
This would fix #2708 and unblock further testing of short timeouts

### CODE REVIEW

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs [hunk 1]
```diff
@@ -999,7 +999,7 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
 
                         if (!suppressEntityBody)
                         {
-                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                            httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
```
Comment (zhenlan) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs]: Just wondering if line 981 should be changed to 
```
httpRequestMessage = _channel.GetHttpRequestMessage(_via);
```

Comment (KKhurin) [reply src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs]: Unfortunately not really :(  _httpRequestMessage is used in a few other code paths that can run simultaneously (and like you probably suspect, that code can run into the same kind of race condition that we're fixing here). So this fix does not completely eliminate all the issues - it just mitigates the most frequent one.

**Comment by StephenBonikowsky** [comment]
@KKhurin If you rebase and force push we can get a clean CI, Matt's PR merged a couple days ago fixes the test failures we are seeing.

**Comment by StephenBonikowsky** [comment]
The test case failures in OSX started before this PR, I think we might have taken an auto-update that caused it. So don't block this PR on those test failures.

LGTM - Merge it!

**Pull request merged**